Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -181,6 +181,54 @@ lemma c_relation : 1728 * W.Δ = W.c₄ ^ 3 - W.c₆ ^ 2 := by
 
 end Quantity
 
+section ModelsWithJInvariant_0_1728
+
+variable (R : Type u) [CommRing R]
+
+/-- The Weierstrass curve $Y^2 + Y = X^3$. -/
+def ofJInvariant0 : WeierstrassCurve R :=
+  ⟨0, 0, 1, 0, 0⟩
+
+lemma ofJInvariant0_c₄ : (ofJInvariant0 R).c₄ = 0 := by
+  simp only [ofJInvariant0, c₄, b₂, b₄]
+  ring1
+
+lemma ofJInvariant0_Δ : (ofJInvariant0 R).Δ = -27 := by
+  simp only [ofJInvariant0, Δ, b₂, b₄, b₆, b₈]
+  ring1
+
+/-- The Weierstrass curve $Y^2 = X^3 + X$. -/
+def ofJInvariant1728 : WeierstrassCurve R :=
+  ⟨0, 0, 0, 1, 0⟩
+
+lemma ofJInvariant1728_c₄ : (ofJInvariant1728 R).c₄ = -48 := by
+  simp only [ofJInvariant1728, c₄, b₂, b₄]
+  ring1
+
+lemma ofJInvariant1728_Δ : (ofJInvariant1728 R).Δ = -64 := by
+  simp only [ofJInvariant1728, Δ, b₂, b₄, b₆, b₈]
+  ring1
+
+end ModelsWithJInvariant_0_1728
+
+section ModelsWithJInvariant
```

Comment:
I think this section and the previous section can be merged. Can you tell me why the particular curves with j-invariant 0 and 1728 are so interesting that they deserve definitions by themselves? Why is it that we care about their `c\_4` and `\Delta` (and not other quantities like `c\_6`)?

Reply:
Sounds good! I think you can remove `section` and `end` as well.

Reply:
> Sounds good! I think you can remove `section` and `end` as well.

But I think this will make the parameter `R` in the `ofJ` explicit?